ee/dtr/admin/configure/external-storage/storage-backend-migration.md

```diff
@@ -0,0 +1,78 @@
+keywords: dtr, storage drivers, local volume, NFS, Azure, S3,
+---
+
+Starting in DTR 2.6, switching storage backends initializes a new metadata store and erases your existing tags. This helps facilitate online garbage collection, which has been introduced in 2.5 as an experimental feature. In earlier versions, DTR would subsequently start a `tagmigration` job to rebuild tag metadata from the file layout in the image layer store. This job has been discontinued, as your storage backend could get out of sync with your DTR metadata, like your manifests and existing repositories. As best practice, DTR storage backends and metadata should always be moved, backed up, and restored together.
+
+## DTR 2.6.4 and above
+
+In DTR 2.6.4, a new flag, `--storage-migrated`, [has been added to `docker/dtr reconfigure`](/reference/dtr/2.6/cli/reconfigure/) which lets you indicate the migration status of your storage data during a reconfigure.  
+
+### Best practice for data migration
+
+Docker recommends the following steps for your storage backend and metadata migration:
+
+1. Disable garbage collection by selecting "Never" under **System > Garbage Collection**, so blobs referenced in the backup that you create continue to exist. See [Garbage collection](/ee/dtr/admin/configure/garbage-collection/) for more details.
+
+    ![](/ee/dtr/images/garbage-collection-0.png){: .img-fluid .with-border}
+
+2. [Back up your existing metadata](/ee/dtr/admin/disaster-recovery/create-a-backup/#back-up-dtr-metadata). See [docker/dtr backup](/reference/dtr/2.6/cli/backup/) for CLI command description and options. 
+
+3. Migrate the contents of your current storage backend to the new one you are switching to. For example, upload your NFS contents to an S3 bucket if you're switching from NFS to S3.
+
+4. [Restore DTR from your backup](/ee/dtr/admin/disaster-recovery/restore-from-backup/) and specify your new storage backend. See [docker/dtr destroy](/reference/dtr/2.6/cli/destroy/) and [docker/dtr restore](/reference/dtr/2.6/cli/backup/) for CLI command descriptions and options.
+
+5. Run garbage collection in "blob" mode. You can either SSH into a DTR node, or [use a UCP client bundle](/ee/ucp/user-access/cli/) to run the following command:
+
+    ```bash
+    docker exec -it dtr-jobrunner-<replica_id> sh
+    ```
+    > See [Find your replica ID](/ee/dtr/admin/disaster-recovery/create-a-backup/#find-your-replica-id) for tips on determining your replica ID.
+
+    Within the running container, type `/bin/job_executor` to start the `job_executor` binary.
+    
+    ```bash
+    /bin/job_executor
+    
+    / # onlinegc blobs
+    ```
+    > Note that the first line results in a display of the binary name, its usage, and available commands including `onlinegc`. To learn more about a command, run `<command_name>` --help.
+
+    Running garbage collection in blob mode destroys any new blobs which are not referenced in your previously created backup.
```

Comment:
Can we add more details about what this means?
Is the purpose of running garbage collection in blob mode to destroy orphaned blobs that may have existed in the old storage backend?

Is there a reason why enabling Online Garbage Collection wouldn't be able to clean these blobs up?
Similar question that I asked before, should Online Garbage Collection not be turned on anymore, or once the user has confirmed that their migration is complete should they turn online GC back on?

Reply:
@caervs can we get your input? I wasn't sure whether we need to share as much detail as possible around blob mode but if I remember Ryan's comment from stand-up, taking care of orphaned blobs should just be part of garbage collection in the future. Right now, when you run `/bin/job_executor`, onlinegc gives you the following options:
```
all       Garbage collect all types
     metadata  Garbage collect image metadata
     joblogs   Garbage collect job logs
     events    Garbage collect events
     blobs     Garbage collect dangling blobs
```

Reply:
Indeed. We should have better user exposure  `blob` mode in a future release.

We need the admin to do this now because Imagine the following user scenario:

```
1. Admin takes back-up
2. Admin disables online gc
3. Admin starts migrating content to new server (this could take hours or even days)
4. Users push new tags and delete old tags
5. Storage migration gets to a blob that is now orphaned but still referenced in the back-up. Thankfully this blob still exists because GC is disabled
6. However storage migration also copies over the new blobs that have been pushed since back-up because it's dumb and just enumerates over all the blobs in the backend
7. So when restore is complete, the admin runs blob mode gc to get rid of the new orphaned blobs
```